compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -752,6 +769,12 @@ class ClassfileParser(
         return unpickleTASTY(in.nextBytes(attrLen))
       }
 
+      if (scan(tpnme.ScalaATTR) && !scalaUnpickleWhitelist.contains(classRoot.name)) {
+        // If the Scala attribute is present but the TASTY attribute isn't, this
+        // classfile is a compilation artifact.
+        return Some(NoEmbedded)
```

Comment:
Don't understand the logic here, What is the significance of ScalaATTR? What happens with Scala2 classfiles?